### ISSUE TITLE
SIRIUS command line with MS1 only data?

### ISSUE THREAD

**Issue by jorainer** [boecker-lab/sirius#25]
We're currently evaluating how to use SIRIUS from/with R and our current workaround (until the REST API is ready - see also #11 ) is to export data from R into a .ms file and call the command line of SIRIUS passing these text files as input. We're trying this at present with MS1 only data (i.e. omitting the `>ms2`) - but the command line SIRIUS complains about the missing MS2 data. How can we run the command line with MS1 data only (if possible at all)?

Regarding the REST API - is there something available what we could test? Ideally something like provide MS1 data and get formulas back?

I'm including also @andreavicini here since he's the one looking into this at present.

**Comment by mfleisch** [comment]
Hey Johannes, 
there is not much SIRIUS can do without MS/MS. 

The only thing that would be possible is decomposing the precursor mass and scoring the resulting formula candidates 
with the isotope pattern scoring. 
All other methods rely on fragmentation trees, which need the MS/MS.

The mass decomposition is already available via the `decomp` standalone sub-command. 
Isotope pattern scoring is currently not available for MS1 only data.

If this is something you would like to use from R we can make this available with the next build.

**Comment by jorainer** [comment]
Thanks for the feedback! 

IMHO it would be great to have this MS1-only formula prediction also available. Less for the *real* annotation processes, but I'm thinking of eventually improving the alignment and/or correspondence analysis in `xcms` based on that (or use it for quality assessment of the grouped chromatographic peaks across samples). We could also use `Rdisop` for that, but would be good to have an alternative - and it would be a first simple use case for the integration of SIRIUS tools into our R workflows.

Is the `decomp` already available or is this what you would make available next build?

**Comment by wkumler** [comment]
I'm jumping in on this thread because I recently figured out this integration and thought I should share. I've been using R with SIRIUS for a while, but only recently updated to the latest version to get `decomp`. Now my formula decomposition workflow looks something like this:

```
mzs <- c(60.081324, 76.039854, 76.076239, 88.987485, 90.055504)

sirius_cmd <- paste0('sirius --noCite decomp',
                     ' --mass=', paste(mzs, collapse = ","),
                     ' --ppm=50',
                     ' --elements=CHNOPS')
sirius_output <- system(sirius_cmd, intern = TRUE)
sirius_output <- sirius_output[!grepl(pattern = "INFO", sirius_output)]
read.table(text = sirius_output, header = TRUE)
```

SIRIUS needs to be added to PATH for this to work, or you can specify the full path to the sirius.exe file. There's a bit of a startup cost to running it this way especially relative to `Rdisop`, so I'll usually pass it as many masses as possible at once.

Also @jorainer I've had some luck using SIRIUS with MS1 only by giving it the base MS1 peak as a single "MS2" scan. The .mgf files I write out from within R look like this:

```
BEGIN IONS
PEPMASS=60.0815010220745
MSLEVEL=1
CHARGE=1+
60.0815010220745 1247721.61759463
END IONS

BEGIN IONS
PEPMASS=60.0815010220745
MSLEVEL=2
CHARGE=1+
60.0815010220745 1247721.61759463
END IONS
```

and you can include additional isotope envelope things in the MS1 and leave the base peak as the only MS2 entry. SIRIUS doesn't seem to complain and I assume this is fine for the purposes of getting a chemical formula from a mass given an isotopic envelope, but someone here is welcome to correct me because I'm unfamiliar with the theory behind the magic.

**Comment by jorainer** [comment]
That sounds great! Thanks for sharing this @wkumler (and sorry for my extremely late reply)! 

Please also have a look at this @andreavicini

**Comment by mfleisch** [comment]
Nice workaround! Just be aware that the `TreeScore` which is part of the `SiriusScore` might be unreliable without MS/MS. 
I would suggest to only use the `IsotopeScore` to rank the candidates.

**Comment by mfleisch** [comment]
With the latest release we introduce a command line parameter (`--allow-ms1-only`) that allows you to import data without MS/MS to run SIRIUS. It handles the TreeScore appropriately and uses only the `IsotopeScore` for ranking if no MS/MS  is available.

**Comment by mfleisch** [comment]
Fixed with version 4.9.0

Feel free to reopen if this does not solve the issue.
